source/code/plugins/in_dsc_monitor.rb

```diff
@@ -85,6 +86,7 @@ def run_check_status
           @dsc_cache[:status] = dsc_status
           @dsc_cache.commit
         }
+        File.chmod(0644, dsc_cache_file)
```

Comment:
Why don't you do chmod right after creating the file in line 28.  why here in check_status ? By doing it here there might be a duration when we create a file with ww and actually update the permissions later on.